abi/abi.go

```diff
@@ -535,6 +540,20 @@ func bigIntToAMDRS(b *big.Int) []byte {
 	return reverse(result[:])
 }
 
+// EcdsaPublicKeyToBytes returns the AMD SEV ABI format of the ECDSA P-384 curve public key.
+func EcdsaPublicKeyToBytes(key *ecdsa.PublicKey) ([]byte, error) {
+	result := make([]byte, EcsdaPublicKeySize)
+	switch key.Curve.Params().Name {
+	case "P-384":
+		binary.LittleEndian.PutUint32(result[0:4], EccP384)
+	default:
+		return nil, fmt.Errorf("ecdsa public key is not on curve P-384")
+	}
+	copy(result[0x04:0x4c], bigIntToAMDRS(key.X))
```

Comment:
Can you add some comments for the numbers here? It can probably be calculated from the EcsdaPublicKeySize struct, but may be easier to understand with a comment, something like "the bytes from 00 to 04 is XXX, 04 to 4c is YYY, 4c to 94 is ZZZ".

Also, maybe a comment to explain why this copy is safe.

Reply:
I'll change the offsets to constants that refer to the API documentation.
These offsets are much smaller than the EcdsaPublicKeySize amount result was created to be just a few lines up. I don't think a comment on the copy safety would help.